be/src/exprs/condition_expr.cpp

```diff
@@ -324,33 +400,72 @@ class VectorizedCoalesceExpr : public Expr {
             return columns[0]->clone();
         }
 
+        if constexpr (lt_is_collection<Type>) {
+            return _evaluate_complex(columns);
+        } else {
+            return _evaluate_general(columns);
+        }
+    }
+
+private:
+    StatusOr<ColumnPtr> _evaluate_general(const Columns columns) {
+        std::vector<ColumnViewer<Type>> viewers;
+        for (auto value : columns) {
+            viewers.push_back(ColumnViewer<Type>(value));
+        }
         // choose not null
         int size = columns[0]->size();
         int col_size = viewers.size();
         ColumnBuilder<Type> builder(size, this->type().precision, this->type().scale);
 
         for (int row = 0; row < size; ++row) {
             int col;
-            for (col = 0; col < (col_size - 1); ++col) {
+            for (col = 0; col < col_size; ++col) {
                 // if not null
                 if (!viewers[col].is_null(row)) {
                     builder.append(viewers[col].value(row));
                     break;
                 }
             }
-
-            // if last column
-            if (col == (col_size - 1)) {
-                if (viewers[col].is_null(row)) {
-                    builder.append_null();
-                } else {
-                    builder.append(viewers[col].value(row));
-                }
+            // if all nulls
+            if (col == col_size) {
+                builder.append_null();
             }
```

Comment:
may append more value

Reply:
`if (col == col_size)` means all columns are null. if the last is not null,` col == col_size -1`.